qiskit/quantum_info/operators/scalar_op.py

```diff
@@ -0,0 +1,264 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+ScalarOp class
+"""
+
+from numbers import Number
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+from qiskit.quantum_info.operators.operator import Operator
+
+
+class ScalarOp(BaseOperator):
+    """Scalar identity operator class.
+
+    This is a symbolic representation of an scalar identity operator on
+    multiple subsystems. It may be used to initialize a symbolic scalar
+    multiplication of an identity and then be implicitly converted to other
+    kinds of operator subclasses by using the :meth:`compose`, :meth:`dot`,
+    :meth:`tensor`, :meth:`expand` methods.
+    """
+
+    def __init__(self, dims, coeff=None):
```

Comment:
```suggestion
    def __init__(self, dims, coeff=1):
```
If you default to 1 here you save a lot of checking for None in the code. Everytime you check for None that's basically a way of saying coefficient is 1.

Reply:
This could be changed